standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -791,10 +799,128 @@ public void rollbackTransaction() {
     }
   }
 
+  @Override
+  public void createCatalog(Catalog cat) throws MetaException {
+    LOG.debug("Creating catalog " + cat.getName());
+    boolean committed = false;
+    MCatalog mCat = catToMCat(cat);
+    try {
+      openTransaction();
+      pm.makePersistent(mCat);
+      committed = commitTransaction();
+    } finally {
+      if (!committed) rollbackTransaction();
+    }
+  }
+
+  @Override
+  public void alterCatalog(String catName, Catalog cat)
+      throws MetaException, InvalidOperationException {
+    if (!cat.getName().equals(catName)) {
+      throw new InvalidOperationException("You cannot change a catalog's name");
```

Comment:
This message is a bit too personal. May be "changing catalog name is not allowed"?